starter-code/src/memory.js

```diff
@@ -1,15 +1,48 @@
-// var MemoryGame = function (cards) {
-//   this.cards = cards;
-// };
+'use strict';
 
-// MemoryGame.prototype.shuffleCard = function (cardsArr) {
+var MemoryGame = function (cards) {
+    this.cards = cards;
+    this.pickedCards = [];
+    this.pairsClicked = 0;
+    this.pairsGuessed = 0;
+};
 
-// };
+MemoryGame.prototype.shuffleCard = function (cardsArr) {
+    var m = cardsArr.length,
+        t, i;
 
-// MemoryGame.prototype.checkIfPair = function (firstCard, secondCard) {
+    // While there remain elements to shuffle…
+    while (m) {
 
-// }
+        // Pick a remaining element…
+        i = Math.floor(Math.random() * m--);
 
-// MemoryGame.prototype.finished = function () {
+        // And swap it with the current element.
+        t = cardsArr[m];
+        cardsArr[m] = cardsArr[i];
+        cardsArr[i] = t;
+    }
 
-// };
+    return cardsArr;
+};
+
+MemoryGame.prototype.checkIfPair = function (firstCard, secondCard) {
+    if (firstCard === secondCard) {
+        this.pairsClicked += 1;
+        this.pairsGuessed += 1;
+        return true;
+    } else {
+        this.pairsClicked += 1;
+        return false;
+    }
+
+}
+
+MemoryGame.prototype.finished = function () {
+    if (this.pairsGuessed === 12) {
```

Comment:
`if (this.pairsGuessed === 12) {`

What if we have a different number of cards?